gson/pom.xml

```diff
@@ -222,4 +238,17 @@
       </plugin>
     </plugins>
   </build>
+  <profiles>
+    <profile>
+      <id>JDK17</id>
+      <activation>
+        <jdk>[17,)</jdk>
+      </activation>
+      <properties>
+        <javaVersion>17</javaVersion>
+        <excludeTestCompilation></excludeTestCompilation>
+        <extraLintSuppressions>,-exports,-missing-explicit-ctor,-removal</extraLintSuppressions>
```

Comment:
I think the cleaner solution would be to have a separate test compilation.

This approach here has the following disadvantages:
- When a JDK 17 is on `PATH` and `JAVA_HOME`, but a `.m2/toolchains.xml` file exists, the current toolchain configuration seems to prefer a JDK 11 (see root `pom.xml`) which then causes build failures because at the same time this profile here is active
- When a JDK 17 is on `PATH` and `JAVA_HOME` this profile causes the build to produce a Gson JAR with target version 17 instead of 7. This also has these other side effects:
  - The JAR file contains an additional `module-info.class` file at the top level
  - `MANIFEST.MF` contains additional OSGi metadata for `Import-Package`, which now seems to define a non-optional dependency on the `java.sql` package
- It erroneously makes Eclipse IDE think the complete project is compiled against JDK 17 (not sure if that can be solved though)

Reply:
I'm not sure exactly what you mean by a separate test compilation?

Reply:
Yes, that's a good idea. Now that we are assuming Java 11+ for the build, we should just be able to set `maven.compiler.release` and `maven.compiler.testRelease`. Though I'm not sure if that's why we're suddenly seeing javadoc failures. :-(

Reply:
> Though I'm not sure if that's why we're suddenly seeing javadoc failures. :-(

Probably previously it was implicit using Java 11 as target because `javadoc` was from JDK 11. Now it uses the `maven.compiler.release` value and fails due to references to `AccessibleObject.canAccess` (added in Java 9) and module issues.

I think the easiest and best solution would be to add `<release>11</release>` (and a short comment explaining the rationale) to the configuration for `maven-javadoc-plugin`. Then we will also get the module overview again in the generated Javadoc.

(You can then probably also revert the Javadoc change for `AccessibleObject#canAccess` again.)

Reply:
Good tips for the javadoc. I tried something else first but now I'm doing what you suggested here and it's looking good. Thanks!